### PR TITLE
Refactor assigned_subtasks

### PR DESCRIPTION
**Pull request by Krigpl**
`assigned_subtasks` was a map holding at most one value, so the refactor is self explanatory.
Golem doesn't support multiple computations at the same time.

### CODE REVIEW

File: golem/monitor/model/taskcomputersnapshotmodel.py [hunk 1]
```diff
@@ -10,4 +10,6 @@ def __init__(self, meta_data, task_computer):
         self.counting_task = task_computer.counting_task
         self.task_requested = task_computer.task_requested
         self.compute_task = task_computer.compute_tasks
-        self.assigned_subtasks = list(task_computer.assigned_subtasks.keys())
+        self.assigned_subtask = ''
+        if task_computer.assigned_subtask:
+            task_computer.assigned_subtask['subtask_id']
```
Comment (maaktweluit) [inline golem/monitor/model/taskcomputersnapshotmodel.py]: what python magic is this? it looks like there is an assignment missing ( or knowledge on my side )

File: golem/task/taskcomputer.py [hunk 2]
```diff
@@ -103,55 +102,52 @@ def __init__(self, task_server: 'TaskServer', use_docker_manager=True,
...
-                    subtask_id,
-                    subtask['docker_images'],
-                    subtask['src_code'],
-                    subtask['extra_data'],
-                    subtask['short_description'],
-                    subtask['deadline'])
-                return True
+        subtask = self.assigned_subtask
+        if not subtask or subtask['task_id'] != task_id:
+            logger.error("Resource colledted for a wrong task, %s", task_id)
```
Comment (maaktweluit) [inline golem/task/taskcomputer.py]: typo, `colledted  => collected`

File: tests/golem/monitor/model/test_taskcomputersnapshotmodel.py [hunk 3]
```diff
@@ -13,7 +13,7 @@ def test_channel(self):
         computer_mock.counting_task = counting_task = random.random() > 0.5
         computer_mock.task_requested = task_requested = random.random() > 0.5
         computer_mock.compute_tasks = compute_tasks = random.random() > 0.5
-        computer_mock.assigned_subtasks = assigned_subtasks = dict((x, None) for x in range(100))
+        computer_mock.assigned_subtask = assigned_subtask = 'test_subtask_id'
```
Comment (maaktweluit) [inline tests/golem/monitor/model/test_taskcomputersnapshotmodel.py]: mock is not matching type, not sure if that matters: `Optional[Dict[str, Any]]`

**Review by maaktweluit — CHANGES_REQUESTED** [review]
LGTM!

Some small comments, will approve on reply!

File: golem/monitor/model/taskcomputersnapshotmodel.py (hunk 1 above)
Comment (Krigpl) [reply golem/monitor/model/taskcomputersnapshotmodel.py]: Good catch, my mistake

File: golem/task/taskcomputer.py [hunk 4]
```diff
@@ -90,8 +90,7 @@ def __init__(self, task_server: 'TaskServer', use_docker_manager=True,
 
         self.stats = IntStatsKeeper(CompStats)
 
-        self.assigned_subtasks = {}
-        self.task_to_subtask_mapping = {}
+        self.assigned_subtask: Dict[str, Any] = {}
```
Comment (Wiezzel) [inline golem/task/taskcomputer.py]: I'd say that empty value should represented as `None` instead of `{}`.

File: golem/task/taskcomputer.py [hunk 5]
```diff
@@ -169,7 +164,8 @@ def task_computed(self, task_thread: TaskThread) -> None:
         work_wall_clock_time = task_thread.end_time - task_thread.start_time
         subtask_id = task_thread.subtask_id
         try:
-            subtask = self.assigned_subtasks.pop(subtask_id)  # ComputeTaskDef
+            subtask = self.assigned_subtask
+            self.assigned_subtask = {}
```
Comment (Wiezzel) [inline golem/task/taskcomputer.py]: See prev. comment.

File: golem/task/taskcomputer.py [hunk 6]
```diff
@@ -402,7 +398,8 @@ def __compute_task(self, subtask_id, docker_images,
                               task_timeout)
         else:
             logger.error("Cannot run PyTaskThread in this version")
-            subtask = self.assigned_subtasks.pop(subtask_id)
+            subtask = self.assigned_subtask
+            self.assigned_subtask = {}
```
Comment (Wiezzel) [inline golem/task/taskcomputer.py]: See prev. comment.

File: golem/task/taskcomputer.py (hunk 4 above)
Comment (Krigpl) [reply golem/task/taskcomputer.py]: I started with that. But then it's easier to do `self.assigned_subtask.get('task_id') == task_id` when it's always a map. I can revert back to it.

Comment (Wiezzel) [reply golem/task/taskcomputer.py]: Maybe add something like this?
```python
@property
def asssigned_task_id(self) -> Optional[str]:
    return self.assigned_subtask.get('task_id') if self.assigned_subtask else None
```
And then just use `self.assigned_task_id == task_id`.

**Review by Wiezzel — APPROVED** [review]

**Review by maaktweluit — APPROVED** [review]

**Pull request merged**